src/components/MapView/Analyser/index.tsx

```diff
@@ -329,21 +327,28 @@ function Analyser({ classes }: AnalyserProps) {
                 >
                   <Typography variant="body2">Download</Typography>
                 </Button>
+                <Button
+                  className={classes.innerAnalysisButton}
+                  onClick={clearAnalysis}
+                >
+                  <Typography variant="body2">Clear Analysis</Typography>
+                </Button>
               </>
             )}
-            <Button
-              className={classes.innerAnalysisButton}
-              onClick={runAnalyser}
-              disabled={
-                (!analysisResult && (!!thresholdError || !selectedDate)) ||
-                !hazardLayerId ||
-                !baselineLayerId
-              }
-            >
-              <Typography variant="body2">
-                {analysisResult ? 'Clear Analysis' : 'Run Analysis'}
-              </Typography>
-            </Button>
+            {!analysisResult && (
+              <Button
+                className={classes.innerAnalysisButton}
+                onClick={runAnalyser}
+                disabled={
+                  (!analysisResult && (!!thresholdError || !selectedDate)) ||
```

Comment:
Everything looks great so far! For `!analysisResult &&` we could probably remove this check since we already condtionally render this component if `analysisResult` is falsey.